internal/bootstrap/bootstrap_tpl.go

```diff
@@ -263,7 +263,9 @@ const bootstrapTemplate = `{
     {{- end }}
   },
   {{- if .StatsSinksJSON }}
-  "stats_sinks": {{ .StatsSinksJSON }},
+  "stats_sinks": [
+    {{ .StatsSinksJSON }}
```

Comment:
Rather than defining this as a single value and worrying about concatenating commas in the go code, couldn't we make this variable a slice of strings and do the `,` joining in the template?

Reply:
Yeah I agree the current state isn't ideal... I'd prefer to park this scope-wise!

I'd have to convert the actual BootstrapConfig `StatsConfigJSON` to a list due to these lines
https://github.com/hashicorp/consul-dataplane/blob/dc0da3cd4ecd2a9e15ed462f5844d7e356ab037d/internal/bootstrap/bootstrap_config.go#L194-L196, where the args have the actual config reconciled. This change would also affect most test and quite a few areas of the code.

My initial thought seeing this package was maybe there might be a better cleanup that could be done here using marshalling/unmarshalling, although this introduces with defining structs/models, it might reduce the code logic complexity and cleanup a most of these handlings in the code